clippy_lints/src/misc_early.rs

```diff
@@ -286,6 +309,20 @@ impl EarlyLintPass for MiscEarlyLints {
                 }
             }
         }
+
+        if let PatKind::Ident(_, ident, Some(ref right)) = pat.node {
+            if let PatKind::Wild = right.node {
+                span_lint(
+                    cx,
+                    REDUNDANT_PATTERN,
+                    pat.span,
+                    &format!(
+                        "the `{} @ _` pattern can be written as just `{}`",
```

Comment:
This seems to be a low hanging fruit to make an auto fixable lint out of this.

Would you like to do this in this PR?

Reply:
I'd like to do that but I don't know how we do (replace with `span_lint_and_sugg`?).

Reply:
Yes exactly. With this lint, this should be relatively self-explanatory. Make sure to read the suggestion rules of rustc here: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.Diagnostic.html#method.span_suggestion

Reply:
applied suggested changes